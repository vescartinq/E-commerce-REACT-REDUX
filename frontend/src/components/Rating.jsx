import React from 'react';

export default function Rating({ rating, numReviews }) {
  return (
    <div className="rating">
      <span>
        <i
          id="star1"
          className={
            rating >= 1
              ? 'fa fa-star'
              : rating >= 0.5
              ? 'fa fa-star-half-o'
              : 'fa fa-star-o'
          }
        />
      </span>
      <span>
        <i
          id="star2"
          className={
            rating >= 2
              ? 'fa fa-star'
              : rating >= 1.5
              ? 'fa fa-star-half-o'
              : 'fa fa-star-o'
          }
        />
      </span>
      <span>
        <i
          id="star3"
          className={
            rating >= 3
              ? 'fa fa-star'
              : rating >= 2.5
              ? 'fa fa-star-half-o'
              : 'fa fa-star-o'
          }
        />
      </span>
      <span>
        <i
          id="star4"
          className={
            rating >= 4
              ? 'fa fa-star'
              : rating >= 3.5
              ? 'fa fa-star-half-o'
              : 'fa fa-star-o'
          }
        />
      </span>
      <span>
        <i
          id="star5"
          className={
            rating >= 5
              ? 'fa fa-star'
              : rating >= 4.5
              ? 'fa fa-star-half-o'
              : 'fa fa-star-o'
          }
        />
      </span>
      <span>{`${numReviews} reviews`}</span>
    </div>
  );
}
