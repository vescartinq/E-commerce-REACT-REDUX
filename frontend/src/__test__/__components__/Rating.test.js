import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import Rating from './../../components/Rating';

describe('Rating', () => {
  let wrapper;

  let rating = 0;
  let numReviews = 0;

  beforeEach(() => {
    wrapper = shallow(<Rating rating={rating} numReviews={numReviews} />);
  });

  test('should render <Rating/> component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should return 0,5 stars', () => {
    rating = 0.5;
    numReviews = 1;

    wrapper = shallow(<Rating rating={rating} numReviews={numReviews} />);

    const i1 = wrapper.find('#star1');

    expect(i1.hasClass('fa fa-star-half-o')).toBeTruthy();
  });

  test('should return 1,5 stars', () => {
    rating = 1.5;
    numReviews = 1;

    wrapper = shallow(<Rating rating={rating} numReviews={numReviews} />);

    const i1 = wrapper.find('#star1');
    const i2 = wrapper.find('#star2');

    expect(i1.hasClass('fa fa-star')).toBeTruthy();
    expect(i2.hasClass('fa fa-star-half-o')).toBeTruthy();
  });

  test('should return 2,5 stars', () => {
    rating = 2.5;
    numReviews = 1;

    wrapper = shallow(<Rating rating={rating} numReviews={numReviews} />);

    const i2 = wrapper.find('#star2');
    const i3 = wrapper.find('#star3');

    expect(i2.hasClass('fa fa-star')).toBeTruthy();
    expect(i3.hasClass('fa fa-star-half-o')).toBeTruthy();
  });

  test('should return 3,5 stars', () => {
    rating = 3.5;
    numReviews = 1;

    wrapper = shallow(<Rating rating={rating} numReviews={numReviews} />);

    const i3 = wrapper.find('#star3');
    const i4 = wrapper.find('#star4');

    expect(i3.hasClass('fa fa-star')).toBeTruthy();
    expect(i4.hasClass('fa fa-star-half-o')).toBeTruthy();
  });

  test('should return 4,5 stars', () => {
    rating = 4.5;
    numReviews = 1;

    wrapper = shallow(<Rating rating={rating} numReviews={numReviews} />);

    const i4 = wrapper.find('#star4');
    const i5 = wrapper.find('#star5');

    expect(i4.hasClass('fa fa-star')).toBeTruthy();
    expect(i5.hasClass('fa fa-star-half-o')).toBeTruthy();
  });

  test('should return 5 stars', () => {
    rating = 5;
    numReviews = 1;

    wrapper = shallow(<Rating rating={rating} numReviews={numReviews} />);

    const i5 = wrapper.find('#star5');

    expect(i5.hasClass('fa fa-star')).toBeTruthy();
  });
});
