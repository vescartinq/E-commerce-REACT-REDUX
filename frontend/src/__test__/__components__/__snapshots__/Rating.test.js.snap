// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render <Rating/> component 1`] = `
<div
  className="rating"
>
  <span>
    <i
      className="fa fa-star-o"
      id="star1"
    />
  </span>
  <span>
    <i
      className="fa fa-star-o"
      id="star2"
    />
  </span>
  <span>
    <i
      className="fa fa-star-o"
      id="star3"
    />
  </span>
  <span>
    <i
      className="fa fa-star-o"
      id="star4"
    />
  </span>
  <span>
    <i
      className="fa fa-star-o"
      id="star5"
    />
  </span>
  <span>
    0 reviews
  </span>
</div>
`;
